PYME/LMVis/layers/octree.py

```diff
@@ -11,23 +11,25 @@
 from OpenGL.GL import *
 
 
+# OCT_SHIFT is just _octant_sign from _octree
+OCT_SHIFT = np.zeros((8,3))
+for n in range(8):
+    OCT_SHIFT[n,0] = 2*(n&1) - 1
+    OCT_SHIFT[n,1] = (n&2) -1
+    OCT_SHIFT[n,2] = (n&4)/2.0 -1
+
 class OctreeRenderLayer(TriangleRenderLayer):
     """
     Layer for viewing octrees. Takes in an octree, splits the faces into triangles, and then uses the rendering engines
     from PYME.LMVis.layers.triangle_mesh.
     """
-    # properties to show in the GUI. Note that we also inherit 'visible' from BaseLayer
-    #vertexColour = CStr('', desc='Name of variable used to colour our points')
-    #cmap = Enum(*cm.cmapnames, default='gist_rainbow', desc='Name of colourmap used to colour faces')
-    #clim = ListFloat([0, 1], desc='How our variable should be scaled prior to colour mapping')
-    #alpha = Float(1.0, desc='Face tranparency')
+    # Additional properties (the rest are inherited from TriangleRenderLayer)
     depth = Int(3, desc='Depth at which to render Octree. Set to -1 for dynamic depth rendering.')
     density = Float(0.0, desc='Minimum density of octree node to display.')
     min_points = Int(10, desc='Number of points/node to truncate octree at')
-    #method = Enum(*ENGINES.keys(), desc='Method used to display faces')
 
-    def __init__(self, pipeline, method='wireframe', dsname='', **kwargs):
-        TriangleRenderLayer.__init__(self, pipeline, method, dsname, **kwargs)
+    def __init__(self, pipeline, method='wireframe', dsname='', context=None, **kwargs):
```

Comment:
Why the change here? In general specifically calling the superclass constructor is preferred (rather than using super()

Reply:
I've recently gotten in the habit of using `super()`. Why is specifically calling the superclass constructor better?

Reply:
Where it really makes a difference is multi-inheritance where the use of `super()` can be ambiguous (or at the very least rather gross), but there are a number of other factors which, IMO, make the explicit form better. These include:
 - different syntax on py2 and py3 making it hard to remember which one to use (or more specifically, the python2  `super(ThisClass, self)` syntax seems needlessly verbose and gross - also needs changing if you change the class name, which seems more likely than changing the base class).
- it doesn't hide what is actually going on under the hood (you are calling the method from the other class on the current object)
- if you change out the base class, it makes you explicitly stop and think about whether the signatures of any over-ridden classes are the same.

This may just be my bias though.